spring-cloud-gcp-data-firestore/src/main/java/org/springframework/cloud/gcp/data/firestore/FirestoreTemplate.java

```diff
@@ -207,12 +231,18 @@ public boolean isUsingStreamTokens() {
 				return Flux.from(instances).doOnNext(t -> writes.add(createUpdateWrite(t)));
 			}
 
-			Flux<List<T>> inputs = Flux.from(instances).bufferTimeout(this.writeBufferSize, this.writeBufferTimeout);
-			return ObservableReactiveUtil.streamingBidirectionalCall(
-					this::openWriteStream, inputs, this::buildWriteRequest);
+			return  Flux.from(instances)
+							.groupBy(t -> getIdValue(t) == null).flatMap(groupedFlux ->
```

Comment:
Minor nit, would prefer if you can format this section so each operator gets its own line like this:

```
return  Flux.from(instances)
							.groupBy(t -> getIdValue(t) == null)
              .flatMap(groupedFlux -> groupedFlux.key() ? create(groupedFlux) : upsert(groupedFlux));
```

Clever usage of Group By!